### ISSUE TITLE
Implement ERC 1167, Clone Factory

### ISSUE THREAD

**Issue by shrugs** [OpenZeppelin/openzeppelin-solidity#1162]
## 🎉 Description

When 1167 is finalized in the new week, it'd be good to have an OpenZeppelin implementation.

https://eips.ethereum.org/EIPS/eip-1167

- [x] 📈 This is a feature request.

**Comment by jbogacz** [comment]
Hey folks, I'd like to take care of this if it's still not touched.

**Comment by nventuro** [comment]
@jbogacz go ahead! I'll gladly review your contribution once it's done :)

**Comment by nventuro** [comment]
After much discussion, we've decided tooling is not yet ready to get something like this into OpenZeppelin. There are too many caveats related to proxies (e.g. constructors, self-destruct) that allow for easy misuse, which go against our design principles. 

Having some sort of way to retrieve the constructor of a contract type at compile time, check for self destruction, etc., would allow for this to be safely implemented, at which point we will reconsider including this in the library.

Closing the issue until such a time comes :stuck_out_tongue:
